packages/core/src/behaviors.ts

```diff
@@ -55,12 +59,31 @@ export function spawnBehavior<TEvent extends EventObject, TEmitted>(
 ): ActorRef<TEvent, TEmitted> {
   let state = behavior.initialState;
   const observers = new Set<Observer<TEmitted>>();
+  const mailbox: TEvent[] = [];
+  let flushing = false;
+
+  const enqueue = (event: TEvent): void => {
+    mailbox.push(event);
+    flush();
+  };
+
+  const flush = () => {
+    if (flushing) {
+      return;
+    }
+    flushing = true;
+    while (mailbox.length > 0) {
+      const event = mailbox.shift()!;
+      state = behavior.transition(state, event, actorCtx);
+      observers.forEach((observer) => observer.next(state));
+    }
+    flushing = false;
+  };
 
   const actor = toActorRef({
     id: options.id,
     send: (event: TEvent) => {
-      state = behavior.transition(state, event, actorCtx);
-      observers.forEach((observer) => observer.next(state));
+      enqueue(event);
```

Comment:
`enqueue` is a super short function - I would probably just inline it here to decrease the number of layers one has to go through to understand how processing works

Reply:
7a91088ba9f53aae846118e214b3091cca842719